File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.*;
class Solution {
    public List<Integer> solution(int[] arr, int[][] queries) {
        List<Integer> ret = new ArrayList<>();
        for(int i = 0; i < queries.length; i++){
            int temp = queries[i][2];
            int min = 1000000;
            for(int j = queries[i][0]; j <= queries[i][1]; j++){
                if(arr[j] > temp){
                    min = Math.min(min, arr[j]);
                }
            }
            if(min == 1000000) ret.add(-1);
            else ret.add(min);
        }
        return ret;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    ret = []
    for q in queries:
        s, e, k = q
        mn = 1000000000
        for i in range(s, e + 1):
            if arr[i] > k:
                mn = min(mn, arr[i])
        ret.append(mn if mn != 1000000000 else -1 )
    return ret